File: python/modelo/personal_salud.py
from abc import abstractmethod
from datetime import date
from modelo.persona import Persona
class PersonalSalud(Persona):
    def __init__(self, dni: str, nombres: str, apellidos: str, fecha_nac: date, sexo: str, 
                 colegiatura: str, especialidad: str, turno: str):
        super().__init__(dni, nombres, apellidos, fecha_nac, sexo)
        self._codigo_colegiatura = colegiatura
        self._especialidad = especialidad
        self._turno_asignado = turno
        self._activo = True
    @abstractmethod
    def firmar_atencion(self, id_atencion: str) -> str:
        pass
    def validar_disponibilidad(self, fecha: date, turno: str) -> bool:
        return self._activo and (self._turno_asignado.upper() == turno.upper())
    def obtener_rol(self) -> str:
        return "PERSONAL_SALUD"

File: java/modelo/PersonalSalud.java
package modelo;
import java.time.LocalDate;
public abstract class PersonalSalud extends Persona {
    protected String codigoColegiatura;
    protected String especialidad;
    protected String turnoAsignado;
    protected boolean activo;
    public PersonalSalud(String dni, String nombres, String apellidos, LocalDate fechaNacimiento,
                         String sexo, String codigoColegiatura, String especialidad, String turnoAsignado) {
        super(dni, nombres, apellidos, fechaNacimiento, sexo);
        this.codigoColegiatura = codigoColegiatura;
        this.especialidad = especialidad;
        this.turnoAsignado = turnoAsignado;
        this.activo = true;
    }
    public abstract String firmarAtencion(String idAtencion);
    public boolean validarDisponibilidad(LocalDate fecha, String turno) {
        return this.activo && this.turnoAsignado.equalsIgnoreCase(turno);
    }
    @Override
    public String obtenerRol() {
        return "PERSONAL_SALUD";
    }
    public String getCodigoColegiatura() {
        return codigoColegiatura;
    }
}